tests/main/snap-icons-fallback/task.yaml

```
@@ -0,0 +1,96 @@
+
+    # Snap which only has a store icon, no local icon file
+    # Local icon: none
+    # Store icon: icon-orange-tail.svg
+    SNAP_FALLBACK=test-snapd-icons-api-fallback
+    SNAP_FALLBACK_ID="$(snap info ${SNAP_FALLBACK} | grep snap-id | tr -s ' ' | cut -d ' ' -f 2)"
+
+    echo "Ensure that snaps don't previously have icons cached"
+    test ! -e "/var/cache/snapd/icons-pool/${SNAP_LOCAL_ID}.icon"
+    test ! -e "/var/cache/snapd/icons-pool/${SNAP_FALLBACK_ID}.icon"
+    test ! -e "/var/cache/snapd/icons/${SNAP_LOCAL_ID}.icon"
+    test ! -e "/var/cache/snapd/icons/${SNAP_FALLBACK_ID}.icon"
+
+    echo "Install a snap which has a local icon and a store icon"
+    snap install "$SNAP_LOCAL"
+
+    echo "Install a snap which has a store icon but no local icon"
+    snap install "$SNAP_FALLBACK"
+
+
+    echo "Check that the snap which is supposed to have a local icon does in fact have a local icon"
+    [ -f "/snap/${SNAP_LOCAL}/current/meta/gui/icon.svg" ]
+    echo "Check that it is the expected icon"
+    diff -a "/snap/${SNAP_LOCAL}/current/meta/gui/icon.svg" icon-orange-chest.svg
+
+    echo "Check that the snap which is NOT supposed to have a local icon does not have no local icon"
+    find /snap/"${SNAP_FALLBACK}"/current/meta/gui/icon.* | wc -l | MATCH 0
+
+
+    echo "Check that both snaps have store icons"
+    snap debug api "/v2/snaps/$SNAP_LOCAL" | MATCH '"icon"'
+    snap debug api "/v2/snaps/$SNAP_FALLBACK" | MATCH '"icon"'
+
+
+    echo "Check that installed snaps now have icons downloaded and installed"
+    test -e "/var/cache/snapd/icons-pool/${SNAP_LOCAL_ID}.icon"
+    test -e "/var/cache/snapd/icons-pool/${SNAP_FALLBACK_ID}.icon"
+    test -e "/var/cache/snapd/icons/${SNAP_LOCAL_ID}.icon"
+    test -e "/var/cache/snapd/icons/${SNAP_FALLBACK_ID}.icon"
+
+    echo "Check that icons download and install directories are only writable by root"
+    # root must have wrx, but otherwise only care that non-root doesn't have write
+    stat --format %A "/var/cache/snapd/icons-pool" | MATCH -e 'drwx.-..-.'
+    stat --format %A "/var/cache/snapd/icons" | MATCH -e 'drwx.-..-.'
+
+    echo "Check that icons downloaded and installed icons are only writable by root"
+    stat --format %A "/var/cache/snapd/icons-pool/${SNAP_LOCAL_ID}.icon" | MATCH -e '-rw-.--.--'
+    stat --format %A "/var/cache/snapd/icons-pool/${SNAP_FALLBACK_ID}.icon" | MATCH -e '-rw-.--.--'
+    stat --format %A "/var/cache/snapd/icons/${SNAP_LOCAL_ID}.icon" | MATCH -e '-rw-.--.--'
+    stat --format %A "/var/cache/snapd/icons/${SNAP_FALLBACK_ID}.icon" | MATCH -e '-rw-.--.--'
+
+
+    # Clear any previously-fetched icons from previous runs
+    rm -f "${SNAP_LOCAL}.icon" "${SNAP_FALLBACK}.icon"
+
+    echo "Check that /v2/icons serves the local icon for the snap which has one"
+    snap debug api "/v2/icons/${SNAP_LOCAL}/icon" > "${SNAP_LOCAL}.icon"
+    diff "${SNAP_LOCAL}.icon" "/snap/${SNAP_LOCAL}/current/meta/gui/icon.svg"
```

the output, in case of differences, won't be necessarily printable, co you can use `cmp` here
```suggestion
    cmp "${SNAP_LOCAL}.icon" "/snap/${SNAP_LOCAL}/current/meta/gui/icon.svg"
```